metals/src/main/scala/scala/meta/internal/builds/BazelBuildTool.scala

```diff
@@ -0,0 +1,184 @@
+package scala.meta.internal.builds
+
+import java.util.concurrent.TimeUnit
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.JavaBinary
+import scala.meta.internal.metals.Messages
+import scala.meta.internal.metals.Messages.ImportBuild
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.Tables
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.internal.process.ExitCodes
+import scala.meta.io.AbsolutePath
+
+import coursierapi.Dependency
+import coursierapi.Fetch
+import org.eclipse.lsp4j.services.LanguageClient
+
+case class BazelBuildTool(
+    userConfig: () => UserConfiguration,
+    projectRoot: AbsolutePath,
+) extends BuildTool
+    with BuildServerProvider
+    with VersionRecommendation {
+
+  override def digest(workspace: AbsolutePath): Option[String] = {
+    BazelDigest.current(projectRoot)
+  }
+
+  def createBspFileArgs(workspace: AbsolutePath): Option[List[String]] =
+    Option.when(workspaceSupportsBsp)(composeArgs())
+
+  def workspaceSupportsBsp: Boolean = {
+    projectRoot.list.exists {
+      case file if file.filename == "WORKSPACE" => true
+      case _ => false
+    }
+  }
+
+  private def composeArgs(): List[String] = {
+    val classpathSeparator = java.io.File.pathSeparator
+    val classpath = Fetch
```

Comment:
Could be a follow up, but it would be woth making sure that this gets downloaded in DownloadDependencies class.